File: sdks/go/pkg/models/operations/getcollaborations.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetCollaborationsStatusEnum string

const (
	GetCollaborationsStatusEnumPending GetCollaborationsStatusEnum = "pending"
)

type GetCollaborationsQueryParams struct {
	Fields []string                    `queryParam:"style=form,explode=false,name=fields"`
	Limit  *int64                      `queryParam:"style=form,explode=true,name=limit"`
	Offset *int64                      `queryParam:"style=form,explode=true,name=offset"`
	Status GetCollaborationsStatusEnum `queryParam:"style=form,explode=true,name=status"`
}

type GetCollaborationsRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetCollaborationsQueryParams
}

type GetCollaborationsResponse struct {
	ClientError    *shared.ClientError
	Collaborations *shared.Collaborations
	ContentType    string
	StatusCode     int64
}

File: sdks/ts/src/sdk/models/operations/getcollaborations.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetCollaborationsStatusEnum {
    Pending = "pending"
}


export class GetCollaborationsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=status" })
  status: GetCollaborationsStatusEnum;
}


export class GetCollaborationsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetCollaborationsQueryParams;
}


export class GetCollaborationsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  collaborations?: shared.Collaborations;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
